### DISCUSSION TITLE
게시물 포스트 API에 관하여

### DISCUSSION THREAD

**Discussion by ljw0096** [prgrms-fe-devcourse/FEDC2_TimeDivider_Root#69] [0. 개발]
# 포스트 Json

현재 앱에서 전송하는 게시물 데이터는 Title, TaskList로 구성되어 있습니다.

따라서 기존 포스트 관련 API 요청을 사용할 때 Custom된 데이터를 기준으로 요청을 주고 받아야 할 것 같습니다.

#### 특정 채널에 포스트 작성하기

특정 채널에 포스트를 작성합니다.

> POST /posts/create

##### Request Header

```
Authorization: bearer JWT토큰
```

##### Request Body

FormData

```
title: String,
image: Binary | null,
channelId: String
```

해당 부분의 title을 Time Divider에 맞게 custom한 내용을 전송해야 할 것 같습니다.

### custom Json
```javascript
const taskPost = {
	title: "Title",
	tasks: [
		{
			id: "ID",
			name: "Something"
		},
		{
			id: "ID",
			name: "Something"
		},
		{
			id: "ID",
			name: "Something"
		}
	]
}

API.createPost(JSON.stringify(taskPost), ...)
```

**Comment by codeisneverodd** [comment]
그러고보니 Time은 넘겨줄 필요가 없군요!
향후 timer 들이 추가, 완료 등 변화가 발생하는 경우 어느 시점에 호출을 해야할지 의논도 해야할 것 같아요!

**Reply by dahye1013** [reply]
그러게요 마냥 useEffect에 넣으면 이것도 API 호출 비용이 과해질 것 같네요.
브라우저가 닫힐때 api 호출하면, api에 대한 성공 여부를 알수 없을 것 같은데.. 🤔
`debounce`나 `throttle` 를 사용하는 건 어때요?

(앗, 서버에 타임을 넘겨주는줄 알고 이야기한건데 시간은 로컬스토리지에 넣고 있어서 안넘기는건가요?)

**Comment by dahye1013** [comment]
전송시 `JSON.stringify`, 송신시 `JSON.parse` 파싱하는 유틸리티를 만들어서
파싱이 필요한 api들에 then에 넣어서 사용하면 좋겠네요 ㅎㅎㅎ!
